File: Raven.Abstractions/Data/GetRequest.cs
using System;
using System.Collections.Generic;
using Raven.Imports.Newtonsoft.Json;

namespace Raven.Abstractions.Data
{
	public class GetRequest
	{
		public string Url { get; set; }
		public IDictionary<string,string> Headers { get; set; }
		public string Query { get; set; }

		[JsonIgnore]
		public string UrlAndQuery
		{
			get
			{
				if (Query == null)
					return Url;
				if (Query.StartsWith("?"))
					return Url + Query;
				return Url + "?" + Query;
			}
		}

		public GetRequest()
		{
			Headers = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase);
		}
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/abstractions/data/GetRequest.java
package net.ravendb.abstractions.data;

import java.util.Map;
import java.util.TreeMap;

import org.codehaus.jackson.annotate.JsonIgnore;

public class GetRequest {
  private String url;
  private Map<String, String> headers = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
  private String query;

  public GetRequest() {
    //empty by design
  }

  public GetRequest(String url) {
    this.url = url;
  }

  public GetRequest(String url, String query) {
    this.url = url;
    this.query = query;
  }

  public Map<String, String> getHeaders() {
    return headers;
  }
  public String getQuery() {
    return query;
  }
  public String getUrl() {
    return url;
  }
  public void setHeaders(Map<String, String> headers) {
    this.headers = headers;
  }
  public void setQuery(String query) {
    this.query = query;
  }
  public void setUrl(String url) {
    this.url = url;
  }

  @JsonIgnore
  public String getUrlAndQuery() {
    if (query == null) {
      return url;
    }
    if (query.startsWith("?")) {
      return url + query;
    }
    return url + "?" + query;
  }

}
